ansible_collections/Juniper/junos_collection/plugins/modules/juniper_junos_system.py

```diff
@@ -285,163 +285,72 @@ def main():
     params = junos_module.params
 
     action = params['action']
-    # Synonymns for shutdown
+    at = params.get('at')
+    in_min = params.get('in_min')
+    all_re = params.get('all_re')
+    other_re = params.get('other_re')
+    media = params.get('media')
+
+    # Synonymn for shutdown
     if action == 'off' or action == 'power_off' or action == 'power-off':
         action = 'shutdown'
 
-    # at option only applies to reboot, shutdown, or halt actions.
-    if (params.get('at') is not None and
-       action != 'reboot' and
-       action != 'shutdown' and
-       action != 'halt'):
-        junos_module.fail_json(msg='The at option can only be used when '
-                                   'the action option has the value "reboot", '
-                                   '"shutdown", or "halt".')
-
-    # in_min option only applies to reboot, shutdown, or halt actions.
-    if (params.get('in_min') is not None and
-       action != 'reboot' and
-       action != 'shutdown' and
-       action != 'halt'):
-        junos_module.fail_json(msg='The in_min option can only be used when '
-                                   'the action option has the value "reboot", '
-                                   '"shutdown", or "halt".')
-
-    # other_re option only applies to reboot, shutdown, or halt actions.
-    if (params.get('other_re') is True and
-       action != 'reboot' and
-       action != 'shutdown' and
-       action != 'halt'):
-        junos_module.fail_json(msg='The other_re option can only be used when '
-                                   'the action option has the value "reboot", '
-                                   '"shutdown", or "halt".')
+    #Four actions are expected - reboot, shutdown, halt and zeroize
+    if action not in {'reboot', 'shutdown', 'halt'}:
```

Comment:
Use List here ['reboot', 'shutdown', 'halt']

Reply:
modified in latest commit.